src/core/Akka/Actor/Internal/ActorSystemImpl.cs

```diff
@@ -383,7 +383,7 @@ public TerminationCallbacks(Task upStreamTerminated)
 
             upStreamTerminated.ContinueWith(_ =>
             {
-                _terminationTask = Interlocked.Exchange(ref _atomicRef, new AtomicReference<Task>(null)).Value;
+                _terminationTask = _atomicRef.GetAndSet(null);
```

Comment:
This was the original reason for the discussion, in the old version it wasn't possible to implement GetAndSet because Interlocked.Exchange is restricted to reference types.